tests/prefetch_related/test_prefetch_related_objects.py

```diff
@@ -58,20 +69,84 @@ def test_m2m_reverse(self):
 
     def test_foreignkey_forward(self):
         authors = list(Author.objects.all())
-        with self.assertNumQueries(1):
+        with self.assertNumQueries(1) as ctx:
             prefetch_related_objects(authors, "first_book")
+            self.assertNotIn("ORDER BY", ctx.captured_queries[-1]["sql"])
 
         with self.assertNumQueries(0):
             [author.first_book for author in authors]
 
+        authors = list(Author.objects.all())
+        with self.assertNumQueries(1) as ctx:
+            prefetch_related_objects(
+                authors,
+                Prefetch(
+                    "first_book",
+                    queryset=Book.objects.order_by("-title"),
+                ),
+            )
+            self.assertNotIn("ORDER BY", ctx.captured_queries[-1]["sql"])
+
     def test_foreignkey_reverse(self):
         books = list(Book.objects.all())
-        with self.assertNumQueries(1):
+        with self.assertNumQueries(1) as ctx:
             prefetch_related_objects(books, "first_time_authors")
+            self.assertIn("ORDER BY", ctx.captured_queries[-1]["sql"])
 
         with self.assertNumQueries(0):
             [list(book.first_time_authors.all()) for book in books]
 
+        books = list(Book.objects.all())
+        with self.assertNumQueries(1) as ctx:
+            prefetch_related_objects(
+                books,
+                Prefetch(
+                    "first_time_authors",
+                    queryset=Author.objects.order_by("-name"),
```

Comment:
I'm not convinced we need to test explicit usage of `order_by` but I'll let mergers decide.

Reply:
From my point of view this is necessary since it corresponds to the case where `queryset.query.order_by` is filled, instead of the case where it is filled later because `queryset.query.default_ordering = True`